File: src/main/java/org/kite9/diagram/dom/processors/xpath/AbstractInlineProcessor.java
package org.kite9.diagram.dom.processors.xpath;

import org.w3c.dom.Element;
import org.w3c.dom.Node;
import org.w3c.dom.Text;

/**
 * Does without copying, changing the elements in place.
 * 
 * @author robmoffat
 *
 */
public abstract class AbstractInlineProcessor extends AbstractProcessor {

	@Override
	public final Node processContents(Node n) {
		return processContents(n, null);
	}

	@Override
	protected Element processTag(Element n) {
		return n;
	}

	@Override
	protected Text processText(Text n) {
		return n;
	}

	@Override
	protected boolean isAppending() {
		return false;
	}
	
	
	
}

File: kite9-visualization/src/commonMain/kotlin/org/kite9/diagram/dom/processors/AbstractInlineProcessor.kt
package org.kite9.diagram.dom.processors

import org.w3c.dom.Comment
import org.w3c.dom.Element
import org.w3c.dom.Node
import org.w3c.dom.Text

/**
 * Does without copying, changing the elements in place.
 *
 * @author robmoffat
 */
abstract class AbstractInlineProcessor : AbstractProcessor() {

    override fun processContents(from: Node?): Node? {
        return processContents(from!!, null)
    }

    override fun processTag(n: Element): Element {
        return n
    }

    override fun processText(n: Text): Text {
        return n
    }

    override val isAppending: Boolean
        protected get() = false

    override fun processComment(c: Comment): Comment {
        return c
    }
}
